rusk-wallet/src/lib/clients.rs

```diff
@@ -0,0 +1,270 @@
+        })?
+        .into_inner()
+        .tx;
+
+        // todo: encode message
+        let msg = PropagateMessage { message: tx_bytes };
+        let req = tonic::Request::new(msg);
+
+        let mut net = self.network.lock().unwrap();
+        let _ = block_in_place(move || {
+            Handle::current().block_on(async move { net.propagate(req).await })
+        })?;
+
+        Ok(())
+    }
+
+    /// Requests an STCT proof.
+    fn request_stct_proof(
+        &self,
+        fee: &Fee,
+        crossover: &Crossover,
+        value: u64,
+        blinder: JubJubScalar,
+        address: BlsScalar,
+        signature: Signature,
+    ) -> Result<Proof, Self::Error> {
+        let mut buf = BytesMut::new();
+        buf.put_slice(&fee.to_bytes());
+        buf.put_slice(&crossover.to_bytes());
+        buf.put_slice(&value.to_bytes());
+        buf.put_slice(&blinder.to_bytes());
+        buf.put_slice(&address.to_bytes());
+        buf.put_slice(&signature.to_bytes());
+
+        let msg = StctProverRequest {
+            circuit_inputs: buf.to_vec(),
+        };
+        let req = tonic::Request::new(msg);
+
+        let mut prover = self.client.lock().unwrap();
+        let res = block_in_place(move || {
+            Handle::current()
+                .block_on(async move { prover.prove_stct(req).await })
+        })?
+        .into_inner()
+        .proof;
+
+        let mut proof_bytes = [0u8; Proof::SIZE];
+        proof_bytes.copy_from_slice(&res);
+
+        let proof = Proof::from_bytes(&proof_bytes)?;
+        Ok(proof)
+    }
+
+    /// Request a WFCT proof.
+    fn request_wfct_proof(
+        &self,
+        commitment: JubJubAffine,
+        value: u64,
+        blinder: JubJubScalar,
+    ) -> Result<Proof, Self::Error> {
+        let mut buf = BytesMut::new();
+        buf.put_slice(&commitment.to_bytes());
+        buf.put_slice(&value.to_bytes());
+        buf.put_slice(&blinder.to_bytes());
+
+        let msg = WfctProverRequest {
+            circuit_inputs: buf.to_vec(),
+        };
+        let req = tonic::Request::new(msg);
+
+        let mut prover = self.client.lock().unwrap();
+        let res = block_in_place(move || {
+            Handle::current()
+                .block_on(async move { prover.prove_wfct(req).await })
+        })?
+        .into_inner()
+        .proof;
+
+        let mut proof_bytes = [0u8; Proof::SIZE];
+        proof_bytes.copy_from_slice(&res);
+
+        let proof = Proof::from_bytes(&proof_bytes)?;
+        Ok(proof)
+    }
+}
+
+/// Implementation of the StateClient trait from wallet-core
+#[derive(Debug)]
+pub struct State {
+    client: Mutex<StateClient<Channel>>,
+}
+
+impl State {
+    pub fn new(client: StateClient<Channel>) -> Self {
+        State {
+            client: Mutex::new(client),
+        }
+    }
+}
+
+/// Types that are clients of the state API.
+impl dusk_wallet_core::StateClient for State {
```

Comment:
Is there a reason you're not importing this, and using the fully qualified path?